scripts/helper.js

```diff
@@ -145,26 +147,29 @@ const GetDepTree = ( name ) => {
 
 
 /**
- * Get all folders within a given path
+ * Get all modules within a given path
+ * The 'module' is a component folder
  *
  * @param  {string}  thisPath - The path that contains the desired folders
  * @param  {boolean} verbose  - Verbose flag either undefined or true
  *
- * @return {array}            - An array of names of each folder
+ * @return {Promise<array>}   - A promise to an array of names of each folder
  */
-const GetFolders = ( thisPath, verbose ) => {
-	try {
-		let folders = Fs.readdirSync( thisPath ).filter(
-				thisFile => Fs.statSync(`${ thisPath }/${ thisFile }`).isDirectory()
-			).filter(
-				thisFile => thisFile !== 'core'
-		);
-
-		return ['core', ...folders ]; // moving core to top
-	}
-	catch( error ) {
-		return [];
-	}
+const GetModules = ( thisPath, verbose ) => {
+	return new Promise(resolve => {
+		Readdir(thisPath)
+			.then(dirContent => {
+				let folders = dirContent
+					.filter(folder => {
+						let path = Path.normalize( `${ thisPath }/${ folder }` );
+						return Fs.statSync( path ).isDirectory() && Fs.existsSync( Path.normalize( `${path}/package.json` ) )
```

Comment:
This function could be refactored to read nicer and is also still synchronous and IO based.